frontend/src/hooks/use-unity-instance.tsx

```diff
@@ -65,28 +65,16 @@ const UnityInstance = () => {
             return;
         }
         const mapContainerBounds = mapContainer.getBoundingClientRect();
-        if (canvasHeight && canvasHeight > 0) {
-            const ratio = mapContainerBounds.width / mapContainerBounds.height;
-            const height = Math.min(mapContainerBounds.height, canvasHeight);
-            canvas.height = height;
-            canvas.width = height * ratio;
-        } else if (canvasHeight === -2) {
-            // "native" ... exactly match the browser/screen pixel density
-            // this will be the highest possible quality, but your GPU
-            // might struggle
-            canvas.height = mapContainerBounds.height * window.devicePixelRatio;
-            canvas.width = mapContainerBounds.width * window.devicePixelRatio;
-        } else {
-            // "auto" ... use the native browser pixel height for
-            // resolution, but never pack more pixels into the canvas than
-            // the actual screen pixels (ie if zoomed out or hiDPI)
-            // quirk: zooming IN will lower the res, which is weird, if you
-            // need to zoom in for some reason probably use "native"
-            canvas.height = Math.min(mapContainerBounds.height * window.devicePixelRatio, mapContainerBounds.height);
-            canvas.width = Math.min(mapContainerBounds.width * window.devicePixelRatio, mapContainerBounds.width);
-        }
+        canvas.height = mapContainerBounds.height * window.devicePixelRatio;
+        canvas.width = mapContainerBounds.width * window.devicePixelRatio;
+        if (unity.sendMessage && g.__globalUnityContext.getCanvasHeight)
+            unity.sendMessage(
+                'ResolutionManager',
+                'SetResoltion',
+                JSON.stringify(g.__globalUnityContext.getCanvasHeight())
+            );
         console.info(`canvas size updated to ${canvas.width}x${canvas.height}`);
-    }, [canvas, canvasHeight]);
+    }, [canvas, unity]);
```

Comment:
```suggestion
    }, [canvas, sendMessage]);
```

since the unity object will never change, this will never trigger an update ... switch to passing in the `sendMessage` func rather than the unity obj to make sure the sendMessage is the one you need

think of like this:

```
const o = {myFunc: () => 1};
const a = o;

o === a // true

o.myFunc = () => 2;

o === a // still true ... changing the myFunc did not change the fact that o is the same object.

Reply:
I see why I don't need unity, but I don't understand why I would to replace it with sendMessage, that wouldn't change either right?

Reply:
it _might_ change ... since it comes from somewhere we do not control, so the hook requires

Reply:
Ah gotcha! So adding the hook ensures that the sendMessage function will still work even if it gets changed somewhere down the road.

Reply:
Linting complains about the useCallback having a missing dependency when I remove the unity hook:
`React Hook useCallback has a missing dependency: 'unity'. Either include it or remove the dependency array`

Should I just keep it in even as well as adding sendMessage, even though it will never change?

Reply:
It's also complaining that sendMessage is unnecessary
`React Hook useCallback has an unnecessary dependency: 'sendMessage'. Either exclude it or remove the dependency array.`

Reply:
you need to not reference the `unity` variable inside the hook ... only reference the actual thing you need (in this case `sendMessage` which is already deconstructed at line ~53)